### DISCUSSION TITLE
why  favorites list are removed from opengrok

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4348] [General]
Our recently released upgraded opengrok instance, which transitions from version 0.10 to 1.12.6, has garnered positive feedback overall. However, there has been a question regarding the removal of the "favorites" feature in the newer version. Additionally, it appears that this new version does not adequately address a specific use case. Favorites page:

<img width="855" alt="Screenshot 2023-05-15 at 2 29 26 PM" src="https://github.com/oracle/opengrok/assets/4188259/7632db33-760d-44d6-be43-b622420d2927">

In the previous version, users could select a set of preferred projects, around 20 for example, and add them to their favorites list. When performing searches for projects, they would then only need to choose one or two projects from their favorites to search for specific text. However, the new approach requires users to manually add or remove projects from the project dropdown every time.

Was there a specific reason why this particular use case was not considered in the new version? Any specific link where this design decision was made to understand pros of this new design.

**Comment by vladak** [comment]
Um, how do you get to this "Favorites" dialog window ? (also, there is curious typo/misspelling in the 'OpenGrok favorities') I am not aware of such thing to exist in OpenGrok. Searching the OpenGrok code base for the strings in the dialog window yields empty results. Is this perhaps a custom modification of yours ?

**Reply by vladak** [reply]
Oh, is this something that was present in 0.10 ? That version is so ancient I don't remember such feature was there. These days the projects are stored in a cookie.

**Reply by jkjha** [reply]
yes... it was 0.10. We were running this version from atleast 6-7 years or more but recently upgraded to the latest one. And so, received complaints from some users about disrupting there workflow for search. The new workflow need them more keystrokes. We have more than 2000+ projects indexed and a user might be interested in only 20 of them. During searches, they would prefer to search within 2-3 specific projects from that subset of 20. We believe this is a valid use case and would appreciate it if this could be taken into consideration during opengrok devepment.

**Reply by vladak** [reply]
Could you describe in detail how it used to work in 0.10 ? I don't remember it.

**Reply by jkjha** [reply]
Sure. Let me try to explain the workflow for first time user:

1. User lands to default search page:

<img width="1402" alt="Screenshot 2023-05-19 at 1 30 36 PM" src="https://github.com/oracle/opengrok/assets/4188259/52523f95-d191-4cca-a4b6-6b3f30f314f9">



2. Now he clicks on "[favorities]" links at bottom of the above list. It opens the below model:

<img width="729" alt="Screenshot 2023-05-19 at 1 37 39 PM" src="https://github.com/oracle/opengrok/assets/4188259/36419f6e-9954-498b-afc2-1913b0dbf4fc">

3. From above models he picks (lets say) 5-6 projects and adds to favorites. Also click the link at bottom of the model: "Show only favorities on search page". 

<img width="757" alt="Screenshot 2023-05-19 at 1 40 45 PM" src="https://github.com/oracle/opengrok/assets/4188259/449f3a6d-8b3b-46c6-94b7-93ab5a5f228c">


4. After pressing Ok on above model, now the default search page shows only Favorites on search page. 

<img width="865" alt="image" src="https://github.com/oracle/opengrok/assets/4188259/325677e5-bd8f-47dc-884f-1a2dcd20b54f">


5. Now he can select any one or 2 projects from his Favorites (example sample project-3 and 4) only and search for any text. He doesn't have to search or look around for all 2000 projects.

Hope this helps. Please feel free to ping if you have any other questions.

**Reply by vladak** [reply]
Okay, this is basically per user project list. The 'Settings' page could be used to reintroduce this feature.

**Comment by vladak** [comment]
I checked out fe80b749cd764b1f0f72b2f913b1fe10581911c3 (0.10 release point, in 2011) to see where the "favourites" code resided so that I can track down what change removed that functionality and more importantly why it was removed, however could not find it (grepping for "Choose projects", "pre-selected", etc.). Are you sure this was not custom modification ?

**Reply by jkjha** [reply]
I apologize sincerely for taking up your valuable time. It appears that our team has made a customized modification, which we discovered after conducting an extensive search. Due to the age of this modification, nobody on our team was even aware of it. Nevertheless, if this feature is relevant and useful, we kindly ask for your consideration.
